test/server/cards/01_SOR/leaders/EmperorPalpatineGalacticRuler.spec.ts

```diff
@@ -0,0 +1,108 @@
+    integration(function(contextRef) {
+        it('Palpatine\'s undeployed ability should deal 1 damage and draw a card at the cost of defeating a friendly, exhausting self, and paying 1 resource', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    leader: 'emperor-palpatine#galactic-ruler',
+                    groundArena: ['battlefield-marine'],
+                    spaceArena: [{ card: 'tie-advanced', upgrades: ['shield'] }],
+                    resources: 5
+                },
+                player2: {
+                    groundArena: ['wampa', 'atst'],
+                }
+            });
+
+            const { context } = contextRef;
+
+            context.player1.clickCard(context.emperorPalpatine);
+            expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.tieAdvanced]);
+            expect(context.player1).not.toHavePassAbilityButton();
+
+            context.player1.clickCard(context.tieAdvanced);
+            expect(context.tieAdvanced).toBeInZone('discard');
+            expect(context.emperorPalpatine.exhausted).toBeTrue();
+            expect(context.player1.exhaustedResourceCount).toBe(1);
+
+            expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.wampa, context.atst]);
+            expect(context.player1).not.toHavePassAbilityButton();
+            context.player1.clickCard(context.atst);
+            expect(context.atst.damage).toBe(1);
+            expect(context.player1.handSize).toBe(1);
+        });
+
+        it('Palpatine\'s undeployed ability should not be able to activate if there are no friendly units', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    leader: 'emperor-palpatine#galactic-ruler',
+                    resources: 5
+                },
+                player2: {
+                    groundArena: ['wampa', 'atst'],
+                }
+            });
+
+            const { context } = contextRef;
+
+            expect(context.emperorPalpatine).not.toHaveAvailableActionWhenClickedBy(context.player1);
+        });
+
+        it('Palpatine\'s on-deploy ability should take control of a damaged unit', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    leader: { card: 'emperor-palpatine#galactic-ruler', exhausted: true },
+                    spaceArena: [{ card: 'tie-advanced', damage: 1 }]
+                },
+                player2: {
+                    groundArena: [{ card: 'wampa', damage: 1 }, 'atst'],
+                }
+            });
+
+            const { context } = contextRef;
+
+            context.player1.clickCard(context.emperorPalpatine);
+            expect(context.player1).toBeAbleToSelectExactly([context.wampa, context.tieAdvanced]);
```

Comment:
Should we consider excluding cards controlled by the Palp player from this ability?

Reply:
I don't think so, we haven't done that for any other ability and I can conceive of game situations where damaging your own units might be beneficial. Plus, in some situations they might be required to damage their own unit if there are no enemy units available, and creating the logic for handling that would be a headache

Reply:
Ah, I thought this was the "take control of" selection, not the deal damage selection